query/src/sql/parser/ast/ast_visitor.rs

```diff
@@ -0,0 +1,394 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_exception::Result;
+
+use crate::sql::parser::ast::BinaryOperator;
+use crate::sql::parser::ast::ColumnDefinition;
+use crate::sql::parser::ast::Expr;
+use crate::sql::parser::ast::Identifier;
+use crate::sql::parser::ast::Indirection;
+use crate::sql::parser::ast::Join;
+use crate::sql::parser::ast::JoinCondition;
+use crate::sql::parser::ast::Literal;
+use crate::sql::parser::ast::OrderByExpr;
+use crate::sql::parser::ast::Query;
+use crate::sql::parser::ast::SQLProperty;
+use crate::sql::parser::ast::SelectStmt;
+use crate::sql::parser::ast::SelectTarget;
+use crate::sql::parser::ast::SetExpr;
+use crate::sql::parser::ast::SetOperator;
+use crate::sql::parser::ast::Statement;
+use crate::sql::parser::ast::TableAlias;
+use crate::sql::parser::ast::TableReference;
+use crate::sql::parser::ast::TypeName;
+use crate::sql::parser::ast::UnaryOperator;
+
```

Comment:
Maybe, there should be annotation here like `PlanVisitor`? @leiysky

Reply:
It's not required, you can suit yourself.